rudder-lang/src/logger.rs

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2019-2020 Normation SAS
+
+use crate::error::*;
+use serde::{Deserialize, Serialize};
+use std::{fmt, str::FromStr};
+
+// to keep LOGS usage as safe as possible, make sure it is only called from this file
+// any outside module call is just a push that happens through macros
+// and a clone to get a safe clone of the global to work on from outside the module
+pub static mut LOGS: Logs = Logs {
+    logs: Vec::new(),
+    log_level: LogLevel::Warn,
+};
+
+pub fn clone_logs() -> Logs {
+    unsafe { LOGS.clone() }
+}
+
+// pub static mut LOGS.log_level: LogLevel = LogLevel::Warn;
+pub fn set_global_log_level(log_level: LogLevel) {
+    unsafe { LOGS.update_log_level(log_level) }
+}
+
+#[macro_export]
+macro_rules! error {
+    ($ ( $ arg : tt ) *) => ({
+        unsafe {
```

Comment:
why is this unsafe ?

Reply:
because `static mut` manipulation can only be done in unsafe blocks

Reply:
so it should be reblaced with a refcell

Reply:
static implies Sync so RefCell won't do the trick, I need to use Mutexes/Arcs